openmp/libomptarget/plugins-nextgen/common/include/GlobalHandler.h

```diff
@@ -58,6 +60,22 @@ class GlobalTy {
   void setPtr(void *P) { Ptr = P; }
 };
 
+typedef void *IntPtrT;
+struct __llvm_profile_data {
+#define INSTR_PROF_DATA(Type, LLVMType, Name, Initializer) Type Name;
+#include "llvm/ProfileData/InstrProfData.inc"
+};
+
+/// PGO profiling data extracted from a GPU device
+struct GPUProfGlobals {
+  std::string names;
+  std::vector<std::vector<int64_t>> counts;
+  std::vector<__llvm_profile_data> data;
+  Triple targetTriple;
+
```

Comment:
These should probably use LLVM structs. E.g. `StringRef` is the name is a constant string with stable storage and `SmallVector`.

I'd really appreciate some descriptions of how this is supposed to look and how it interacts with the existing profile data.

Reply:
The basic idea is that the fields in this struct are going to be passed to compiler-rt to be made into a profraw file for this specific target.

I don't know if `llvm::StringRef` would work for the names field. The names data is constant on the GPU device, but dynamic data still has to be allocated on the host in order to read the GPU value. I think it makes the most sense to make the struct own the name data. 

Do you think I should change all the vectors to `SmallVector` or just the innermost `Counts` vector?

Reply:
All of them, SmallVector is a std::vector with small size optimizations like `std::string` basically.

Reply:
I can't seem to use `SmallVector` for `__llvm_profile_data` because many of its defined fields are const qualified and it needs an assignment operator definition. I'm able to use `std::remove_const`, but that feels kind of hack-y. Not sure if the best solution is to use `std::vector`, `std::remove_const`, or remove the const qualifiers in [InstrProfData.inc](https://github.com/llvm/llvm-project/blob/main/llvm/include/llvm/ProfileData/InstrProfData.inc).

Reply:
That's confusing, how would using a `std::vector` not have that problem as well? I'll need to look into that.